### ISSUE TITLE
azurerm_static_site fails to create Static Site on azurerm versions 2.93.0 and above

### ISSUE THREAD

**Issue by piotr-lasota** [hashicorp/terraform-provider-azurerm#15216]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v1.1.4
on linux_amd64 (same on darwin_amd64)
+ provider registry.terraform.io/hashicorp/azurerm v2.93.0
```
### Affected Resource(s)

* `azurerm_static_site` versions above and including `2.93.0`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "2.93.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "group" {
  name     = "rg-issue-with-azurerm-static-site"
  location = "westeurope"
}

resource "azurerm_static_site" "site" {
  name                = "app-that-will-fail-to-get-created"
  resource_group_name = azurerm_resource_group.group.name
  location            = azurerm_resource_group.group.location
  sku_size            = "Free"
  sku_tier            = "Free"
}
```

### Debug Output

[Debug output Github Gist link](https://gist.github.com/piotr-lasota/40d4f8b02b33a08cec85e2aa8bd37765)

### Panic Output

Not applicable

### Expected Behaviour

`azurerm_static_site` gets created

### Actual Behaviour
Creating the resource fails
```
azurerm_static_site.website: Creating...
╷
│ Error: failed creating Static Site: (Name "app-that-will-fail-to-get-created" / Resource Group "rg-issue-with-azurerm-static-site"): web.StaticSitesClient#CreateOrUpdateStaticSite: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="SkuCode 'SKU' is invalid." Details=[{"Message":"SkuCode 'SKU' is invalid."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","ExtendedCode":"55925","Message":"SkuCode 'SKU' is invalid.","MessageTemplate":"SkuCode '{0}' is invalid.","Parameters":["SKU","Free"]}}]
│ 
│   with azurerm_static_site.website,
│   on main.tf line 19, in resource "azurerm_static_site" "website":
│   19: resource "azurerm_static_site" "website" {
│ 
╵
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- does not reproduce on `2.92.0`
- does reproduce on
  - `2.93.0`
  - `2.93.1`
  - `2.94.0`

`BadRequest` details include
```
Error: Code="BadRequest"
Message="SkuCode 'SKU' is invalid."
Details=[{"Message":"SkuCode 'SKU' is invalid."}
```
this looks like if instead of `"Free"` the value passed in the request was `"SKU"`.

### References

Searching for similiar issue yielded no results so I assume this one is first.

**Comment by tombuildsstuff** [comment]
Duplicate of #15125 which has been fixed in #15141 which will ship in v2.95.0

**Comment by piotr-lasota** [comment]
@tombuildsstuff sorry for missing that. I only searched in open issue, my bad! (TIL: check closed issues as well when looking for references)
